src/cascadia/TerminalApp/LeafPane.cpp

```diff
@@ -0,0 +1,717 @@
+            auto child = winrt::get_self<winrt::TerminalApp::implementation::TerminalSettings>(_control.Settings());
+            auto parent = winrt::get_self<winrt::TerminalApp::implementation::TerminalSettings>(settings);
+            child->ClearParents();
+            child->InsertParent(0, parent->get_strong());
+            _control.UpdateSettings();
+        }
+    }
+
+    // Method Description:
+    // - Focuses this pane if the given id matches ours
+    // Arguments:
+    // - The ID of the pane we want to focus
+    void LeafPane::FocusPane(uint32_t id)
+    {
+        if (_id == id)
+        {
+            _control.Focus(FocusState::Programmatic);
+        }
+    }
+
+    // Method Description:
+    // - Focuses this control
+    void LeafPane::FocusFirstChild()
+    {
+        if (Root().ActualWidth() == 0 && Root().ActualHeight() == 0)
+        {
+            // When these sizes are 0, then the pane might still be in startup,
+            // and doesn't yet have a real size. In that case, the control.Focus
+            // event won't be handled until _after_ the startup events are all
+            // processed. This will lead to the Tab not being notified that the
+            // focus moved to a different Pane.
+            //
+            // In that scenario, trigger the event manually here, to correctly
+            // inform the Tab that we're now focused.
+            _GotFocusHandlers(*this);
+        }
+
+        _control.Focus(FocusState::Programmatic);
+    }
+
+    // Method Description:
+    // - Returns true if this pane is currently focused
+    // Return Value:
+    // - true if the currently focused pane is this pane
+    bool LeafPane::HasFocusedChild()
+    {
+        return _control && _lastActive;
+    }
+
+    bool LeafPane::ContainsReadOnly()
+    {
+        return _control.ReadOnly();
+    }
+
+    // Method Description:
+    // - Splits this pane, creating a new leaf pane and a parent pane
+    // - The parent pane holds this pane and the newly created neighbour
+    // - Emits an event with the new parent, so that whoever is listening
+    //   will replace us with our parent
+    // Arguments:
+    // - splitType: what type of split we want to create.
+    // - profile: The profile GUID to associate with the newly created pane.
+    // - control: A TermControl to use in the new pane.
+    // Return Value:
+    // - The newly created pane
+    TerminalApp::LeafPane LeafPane::Split(winrt::Microsoft::Terminal::Settings::Model::SplitState splitType,
+                                          const float splitSize,
+                                          const GUID& profile,
+                                          const winrt::Microsoft::Terminal::TerminalControl::TermControl& control)
+    {
+        splitType = _convertAutomaticSplitState(splitType);
+        auto newNeighbour = TerminalApp::LeafPane(profile, control, false);
```

Comment:
```suggestion
        auto newNeighbour{ make<LeafPane>(profile, control, false)};
```
nit: prefer winrt::make/make_self